src/Stackctl/StackSpec.hs

```diff
@@ -135,11 +135,29 @@ writeTemplateBody path body = do
   dir = takeDirectory path
   ext = takeExtension path
 
-writeStackSpec :: MonadUnliftIO m => StackSpec -> TemplateBody -> m ()
-writeStackSpec stackSpec templateBody = do
-  writeTemplateBody templatePath templateBody
-  createDirectoryIfMissing True $ takeDirectory specPath
-  liftIO $ Yaml.encodeFile specPath $ stackSpecSpecBody stackSpec
+writeStackSpec
+  :: (MonadUnliftIO m, MonadLogger m)
+  => Bool
+  -> StackSpec
+  -> Maybe TemplateBody
+  -> m ()
+writeStackSpec overwrite stackSpec mTemplateBody = do
+  for_ mTemplateBody $ \templateBody -> do
+    logInfo $ "Writing template" :# ["path" .= templatePath]
+    writeTemplateBody templatePath templateBody
```

Comment:
We always (over)write a template if one was given. The `overwrite` argument is solely related to the specification file. This mismatch between `(overwrite? + content)` for spec and `(nothing | just content)` for template makes me a little uncomfortable, but I think addressing that in the types would border on overkill. Everything should be going through `Generate`, making this function's interface less important.